zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/callback/ZkAsyncCallbacks.java

```diff
@@ -111,12 +112,17 @@ protected void recordFailure(int rc, String path, ZkAsyncCallMonitorContext moni
     }
   }
 
-  public static class CreateCallbackHandler extends DefaultCallback implements StringCallback {
+  public static class CreateCallbackHandler extends DefaultCallback implements StringCallback, Create2Callback {
```

Comment:
The reason is the callback type in Zookeeper which supports TTL mode is of type `Create2Callback` (see [here](https://github.com/apache/zookeeper/blob/master/zookeeper-server/src/main/java/org/apache/zookeeper/ZooKeeper.java#L1602)). So I either had to add another `CallbackHandler` type that implements `Create2Callback`, or add `Create2Callback` to `CreateCallbackHandler` to support async create operations with TTL support. Adding a new type resulted in more duplicate code of the same methods with different argument types downstream, so I decided to just have `CreateCallbackHandler` implement `Create2Callback` instead.

Reply:
Makes sense. I was expecting to see a new `Create2CallbackHandler`, but since we're not doing much other than recording some stats in the monitor, I'm fine with this.